server/presenters/document.js

```diff
@@ -1,24 +1,46 @@
 // @flow
 import { takeRight } from 'lodash';
-import { User, Document } from '../models';
+import { User, Document, Attachment } from '../models';
+import { getSignedImageUrl } from '../utils/s3';
 import presentUser from './user';
 
 type Options = {
   isPublic?: boolean,
 };
 
+const attachmentRegex = /!\[.*\]\(\/api\/attachments\.redirect\?id=(?<id>.*)\)/gi;
+
+// replaces attachments.redirect urls with signed/authenticated url equivalents
+async function replaceImageAttachments(text) {
```

Comment:
Not sure if it's worth putting this utility method elsewhere? It's only used here so seems to make sense to co-locate.

Reply:
it is pretty much encapsulated perfectly in here. No other "unit" is publicly shareable except for Documents, so I think it's good here. Also definitely much neater to have it in the `presenter` module.